pkg/ceph/osd/agent_test.go

```diff
@@ -567,31 +368,32 @@ func TestGetPartitionPerfSchemeDiskInUse(t *testing.T) {
 	}
 	defer os.RemoveAll(configDir)
 
-	etcdClient := util.NewMockEtcdClient()
 	context := &clusterd.Context{
-		DirectContext: clusterd.DirectContext{EtcdClient: etcdClient, NodeID: "a", Inventory: createInventory()},
-		ConfigDir:     configDir,
+		Devices:   []*clusterd.LocalDisk{},
+		ConfigDir: configDir,
 	}
 
 	a := &OsdAgent{desiredDevices: []string{"sda"}, kv: kvstore.NewMockKeyValueStore()}
 
 	// mock device sda having been already partitioned
 	_, sdaUUID := mockPartitionSchemeEntry(t, 1, "sda", nil, a.kv, a.nodeName)
 
-	context.Inventory.Local.Disks = []*inventory.LocalDisk{
+	context.Devices = []*clusterd.LocalDisk{
 		{Name: "sda", Size: 107374182400, UUID: sdaUUID}, // 100 GB
 	}
 
-	// mock device sda already being saved to desired state
-	etcdClient.SetValue(fmt.Sprintf("/rook/services/ceph/osd/desired/a/device/%s/osd-id-data", sdaUUID), "1")
-	etcdClient.SetValue(fmt.Sprintf("/rook/services/ceph/osd/desired/a/device/%s/osd-id-metadata", sdaUUID), "1")
+	//etcdClient.SetValue(fmt.Sprintf("/rook/services/ceph/osd/desired/a/device/%s/osd-id-data", sdaUUID), "1")
+	//etcdClient.SetValue(fmt.Sprintf("/rook/services/ceph/osd/desired/a/device/%s/osd-id-metadata", sdaUUID), "1")
 
 	// load desired devices, this should return that sda is desired to have osd 1
 	devices, err := a.loadDesiredDevices(context)
```

Comment:
i don't think that `loadDesiredDevices()` is called from any product code anymore.  All usage of it in the tests should be removed.  This test here is still valid, but we shouldn't be calling loadDesiredDevices at all, I think just the call to `getPartitionPerfScheme` below is important.

Some other tests may be in a similar boat since `loadDesiredDevices` is called in a few other tests.  Feel free to chat with me about those specific tests too once you have removed `loadDesiredDevices` from the codebase.

Reply:
great, i'll remove that thanks